### ISSUE TITLE
Environment variable values not overwriting config file values

### ISSUE THREAD

**Issue by DominykasPetke** [flightlessmango/MangoHud#444]
When using `MANGOHUD_CONFIG=read_cfg` and then trying to use a setting which already existed in the config file, the environment variable's option is not used.

Example: 
Config file has `position=top-left`
`MANGOHUD_CONFIG="read_cfg,position=top-right" mangohud vkcube`

Expected result: HUD is in the top-right corner
Actual result: HUD is in the top-left corner

This is broken since 0.6.0, 0.5.1 works as expected.

**Comment by gnusenpai** [comment]
If you are (or anyone reading this is) willing to compile your own MangoHud, here is a one-liner patch that allows non-element-affecting parameters to be used in conjunction with `MANGOHUD_CONFIG=read_cfg` with `legacy_layout=0` in MangoHud.conf:
```diff
From 3c040f8a5069e0a6aa55be04670027f484f17ddb Mon Sep 17 00:00:00 2001
From: gnusenpai <pspslimboy98@gmail.com>
Date: Thu, 3 Mar 2022 17:43:40 -0500
Subject: [PATCH] fix read_cfg with legacy_layout disabled

---
 src/overlay_params.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/overlay_params.cpp b/src/overlay_params.cpp
index 5818c45..2ebf9d7 100644
--- a/src/overlay_params.cpp
+++ b/src/overlay_params.cpp
@@ -697,7 +697,7 @@ parse_overlay_config(struct overlay_params *params,
 
    // TODO decide what to do for legacy_layout=0
    // second pass, override config file settings with MANGOHUD_CONFIG
-   if (params->enabled[OVERLAY_PARAM_ENABLED_legacy_layout] && env && read_cfg) {
+   if (env && read_cfg) {
       // If passing legacy_layout=0 to MANGOHUD_CONFIG anyway then clear first pass' results
       HUDElements.ordered_functions.clear();
       parse_overlay_env(params, env);
-- 
2.34.1
```
I can't promise that it won't break things if you try to add/remove elements with `MANGOHUD_CONFIG`... :stuck_out_tongue:
However, it should still be useful for things like `position` or `fps_limit` and such.

**Comment by QmwJlHuSg9pa** [comment]
I'm running 0.6.9. I'm not sure what intended behaviour is for next release, but here's my request for precedence:

Given:

`MANGOHUD_CONFIG=no_display,read_cfg,fps_limit=30`

- If valid config file is present: ignore all options _before_ "read_cfg", then process options within "read_cfg", then process all following options. I.e reduced to: `read_cfg,fps_limit=30`
- If _no_ valid config file is present: process all options before "read_cfg", ignore "read_cfg", then process all following options. I.e. reduced to: `no_display,fps_limit=30`.

On a related note, how does MangoHud handle `read_cfg` in conjunction with `preset` -- is there potential for recursive loops? E.g. regular config containing "preset" option, which points to a preset config containing "read_cfg"?. Perhaps related to #845.

**Comment by laribA** [comment]
Please reopen, issue is still present today—on normal and Flatpak MangoHud. I had to apply @gnusenpai's one-liner fix manually when compiling MangoHud, else it wouldn't work, and I don't know of any way to fix this with Flatpak.

**Comment by Oikio** [comment]
Can confirm on  `v0.7.1` 

Assuming config contains `fps_limit=144,120,90,60`,  `MANGOHUD_CONFIG=read_cfg,fps_limit=90` will overriding it.

**Comment by cipitaua** [comment]
confirming the issue on v0.7.1, MANGOHUD_CONFIG seems mostly ignored. MANGOHUD_CONFIGFILE does work instead.

**Comment by flightlessmango** [comment]
There's no value in finding bugs in an old version of mangohud.
At least use the latest version or preferably use the latest git

**Comment by cipitaua** [comment]
> There's no value in finding bugs in an old version of mangohud. At least use the latest version or preferably use the latest git

you're right. I've downloaded the 0.7.2 binaries, and tried
```
LD_LIBRARY_PATH="$PWD/lib/" MANGOHUD_CONFIG=fps,resolution $PWD/bin/mangohud vkcube
```

attached is the result

![image](https://github.com/user-attachments/assets/b32b8da7-1c53-4435-a810-61e9c3a66edc)

**Comment by Kyr4l** [comment]
may i suggest reopening this issue ?

**Comment by flightlessmango** [comment]
@Kyr4l do you experience the same bug with mangohud 0.7.2 or later?

**Comment by Kyr4l** [comment]
> @Kyr4l do you experience the same bug with mangohud 0.7.2 or later?

![image](https://github.com/user-attachments/assets/0705564c-87a6-4478-aa98-f8d71ad99a15)
yes

**Comment by flightlessmango** [comment]
The way you're testing 0.7.2 wouldn't work, that would still use installed version which I presume is 0.7.1

**Comment by Kyr4l** [comment]
> The way you're testing 0.7.2 wouldn't work, that would still use installed version which I presume is 0.7.1

i've called ./mangohud which is the directory i'm in currently, below i've called it with an absolute path instead
![image](https://github.com/user-attachments/assets/0ff153e7-5ebe-4a5c-99e2-91de4d51554f)

![image](https://github.com/user-attachments/assets/1e3e5ef0-2063-42ed-854a-bde2b15f065f)

**Comment by flightlessmango** [comment]
That's not how the script is designed or how the vulkan loader functions. It won't work

**Comment by Kyr4l** [comment]
> That's not how the script is designed or how the vulkan loader functions. It won't work



> That's not how the script is designed or how the vulkan loader functions. It won't work

noted

so instead i've installed it on a debian install
here's the result
![image](https://github.com/user-attachments/assets/2e09430b-5057-4ea3-b980-a1cdf6012236)

![image](https://github.com/user-attachments/assets/9b1cda55-517f-411d-a47b-565dca0173af)

**Comment by flightlessmango** [comment]
Can you then verify that `mangohud --version` is 0.7.2?

**Comment by Kyr4l** [comment]
> Can you then verify that `mangohud --version` is 0.7.2?

it is, i've installed the latest version from the releases tab
![image](https://github.com/user-attachments/assets/d3af0daa-5c7c-40f1-aa3d-ad938a7492eb)

**Comment by flightlessmango** [comment]
I can't repro this on 0.7.2, it *should* work

**Comment by Kyr4l** [comment]
> I can't repro this on 0.7.2, it _should_ work

i only have a very basic MangoHud.conf

![image](https://github.com/user-attachments/assets/4dcba09f-0ada-42f5-ba23-8108976cec9f)

as you can see the fps_limit is set to 20, and that's indeed what happens when i launch glxgears, the fps get limited to 20, but when i add the config fps_limit=10 in the laungh args + the read_cfg variable, the 10 fps max limit doesn't override the config value.

did you try to replicate the same thing ?

**Comment by flightlessmango** [comment]
I can repro it with `legacy_layout=false` on 0.7.2, but it appears to have been fixed already after 0.7.2

**Comment by Kyr4l** [comment]
good to know, thanks!
